webpack/scenes/ContentViews/Details/Repositories/__tests__/contentViewAddRemove.test.js

```diff
@@ -0,0 +1,135 @@
+const repoTypesResponse = [{ name: 'deb' }, { name: 'docker' }, { name: 'file' }, { name: 'ostree' }, { name: 'puppet' }, { name: 'yum' }];
+const cvDetailData = require('../../__tests__/contentViewDetails.fixtures.json');
+
+const cvDetailsPath = api.getApiUrl('/content_views/1');
+const cvRepoAddData = require('./contentViewRepoAdd.fixture.json');
+
+let firstRepo;
+let searchDelayScope;
+let autoSearchScope;
+
+beforeEach(() => {
+  const { results } = repoData;
+  [firstRepo] = results;
+  searchDelayScope = mockSetting(nockInstance, 'autosearch_delay', 500);
+  autoSearchScope = mockSetting(nockInstance, 'autosearch_while_typing', true);
+  nockInstance
+    .persist() // match any query to this endpoint, gets cleaned up by `nock.cleanAll()`
+    .get(api.getApiUrl('/repositories/repository_types'))
+    .query(true)
+    .reply(200, repoTypesResponse);
+  nockInstance
+    .persist()
+    .get(cvDetailsPath)
+    .query(true)
+    .reply(200, cvDetailData);
+});
+
+afterEach(() => {
+  nock.cleanAll();
+  assertNockRequest(searchDelayScope);
+  assertNockRequest(autoSearchScope);
+});
+
+test('Can enable and disable add repositories button', async (done) => {
+  const autocompleteScope = mockAutocomplete(nockInstance, autocompleteUrl);
+
+  const scope = nockInstance
+    .get(cvAllRepos)
+    .query(true)
+    .reply(200, repoData);
+
+  const { getByText, getByLabelText } = renderWithRedux(
+    <ContentViewRepositories cvId={1} />,
+    renderOptions,
+  );
+
+  await patientlyWaitFor(() => expect(getByText(firstRepo.name)).toBeTruthy());
+  expect(getByLabelText('Select all rows')).toBeInTheDocument();
+  expect(getByLabelText('add_repositories')).toHaveAttribute('aria-disabled', 'true');
+  getByLabelText('Select all rows').click();
+  await patientlyWaitFor(() => {
+    expect(getByLabelText('add_repositories')).toHaveAttribute('aria-disabled', 'false');
+  });
+  assertNockRequest(autocompleteScope);
+  assertNockRequest(scope, done);
+});
+
+test('Can add repositories', async (done) => {
+  const autocompleteScope = mockAutocomplete(nockInstance, autocompleteUrl);
+  const cvAddparams = { repository_ids: [58, 62, 64, 107, 106] };
+
+  const repoAddscope = nockInstance
+    .put(cvDetailsPath, cvAddparams)
+    .reply(200, cvRepoAddData);
+
+  const scope = nockInstance
+    .get(cvAllRepos)
+    .query(true)
+    .reply(200, repoData);
+
+  const { getByText, getByLabelText } = renderWithRedux(
+    <ContentViewRepositories cvId={1} />,
+    renderOptions,
+  );
+
+  await patientlyWaitFor(() => expect(getByText(firstRepo.name)).toBeTruthy());
+  expect(getByLabelText('Select all rows')).toBeInTheDocument();
+  expect(getByLabelText('add_repositories')).toHaveAttribute('aria-disabled', 'true');
+  getByLabelText('Select all rows').click();
+  getByLabelText('add_repositories').click();
+  assertNockRequest(autocompleteScope);
+  assertNockRequest(scope);
+  assertNockRequest(repoAddscope);
+  assertNockRequest(scope, done);
```

Comment:
you shouldn't have to assert the same scope twice, you can keep the one with the `done` only, same for the others in this file